fileHub/js-app/tests/app/services/tests-router.js

```diff
@@ -7,46 +7,64 @@ const REGISTER_ROUTE = 'register';
 const ERROR_ROUTE = '404';
 
 module('Router service', (hooks) => {
-  test('Should show default page when hash is correctly', (assert) => {
+  test('Should show default page when hash is correct', (assert) => {
     const testWindow = {
       location: {hash: '#register'},
 
       addEventListener(event, init) {
-        init();
       },
     };
 
     const configuration = getConfig(assert);
     new Router(configuration, testWindow);
-    assert.verifySteps([REGISTER_ROUTE, REGISTER_ROUTE], 'Should route to default page.');
+    assert.verifySteps([REGISTER_ROUTE], 'Should route to register page.');
   });
 
   test('Should show default page when hash is empty', (assert) => {
     const testWindow = {
       location: {hash: '#'},
 
       addEventListener(event, init) {
-        init();
       },
     };
 
     const configuration = getConfig(assert);
     new Router(configuration, testWindow);
-    assert.verifySteps([DEFAULT_ROUTE, DEFAULT_ROUTE], 'Should route to default page.');
+    assert.verifySteps([DEFAULT_ROUTE], 'Should route to default page.');
   });
 
   test('Should show error page when hash is invalid', (assert) => {
     const testWindow = {
       location: {hash: '#router456'},
 
       addEventListener(event, init) {
-        init();
       },
     };
 
     const configuration = getConfig(assert);
     new Router(configuration, testWindow);
-    assert.verifySteps([ERROR_ROUTE, ERROR_ROUTE], 'Should route to error page.');
+    assert.verifySteps([ERROR_ROUTE], 'Should route to error page.');
+  });
+
+  test('Should router works correctly when hash is changed', (assert) => {
+    const configuration = getConfig(assert);
+
+    const testWindow = {
+      location: {hash: '#'},
+
+      addEventListener(event, init) {
+        if (event === 'hashChange') {
+          const step = configuration.getPageByHash(init);
+          step();
```

Comment:
This should not be inside of the mock. Let's have a mock window which extends EventTarget or imitates event target.